Show discard or keep alert after tapping outside of the PR reviewers,…

julienbodet opened the pull request:
… assignees, milestones, labels modal

This my first contribution to GitHawk :) Thanks for the awesome project!

I implemented the feature from #2582. Now after tapping outside the modal for PR reviewers, assignees, milestone or labels, an alert appears asking to keep or discard the changes, if any.

BasThomas commented on Classes/Issues/IssueManagingContextController.swift:
```diff
@@ -323,10 +335,22 @@ final class IssueManagingContextController: NSObject, ContextMenuDelegate {
         let selected = controller.selected
         guard controller.selectionChanged(newValues: selected) else { return }
+
+        if controller.wasDismissedByDone {
```
Can we put this logic in a function, as it seems to be repeated from lines 312-320?

BasThomas commented on Classes/Issues/IssueManagingContextController.swift:
```diff
@@ -363,10 +399,24 @@ final class IssueManagingContextController: NSObject, ContextMenuDelegate {
+        guard let viewController = self.viewController else { return }
+
+        let title = NSLocalizedString("Are you sure?", comment: "")
+        let message = NSLocalizedString("Some changes have been made. Are you sure to discard them?", comment: "")
```
`Are you sure you want to discard them?` might be a bit more pleasing to read, I think? :)

julienbodet replied:
Done

BasThomas commented on Classes/Labels/LabelsViewController.swift:
```diff
@@ -19,6 +19,8 @@ LabelSectionControllerDelegate {
     private let client: GithubClient
     private let request: RepositoryLabelsQuery
 
+    var wasDismissedByDone: Bool = false
```
The `: Bool` is inferred; you can omit it!

julienbodet replied:
Done

BasThomas commented on Classes/Issues/IssueManagingContextController.swift:
```diff
@@ -363,10 +399,24 @@ final class IssueManagingContextController: NSObject, ContextMenuDelegate {
 
     func contextMenuDidDismiss(viewController: UIViewController, animated: Bool) {}
 
+    func showCancelAlert(onKeep: @escaping () -> Void) {
```
What about `completion` rather than `onKeep`, which provides a `Bool` result? Reading this I'm wondering if "on keep" is easy to understand.

julienbodet replied:
Agreed!

BasThomas commented on Classes/People/PeopleViewController.swift:
```diff
@@ -30,6 +30,8 @@ PeopleSectionControllerDelegate {
     private var owner: String
     private var repo: String
 
+    var wasDismissedByDone: Bool = false
```
Should this variable be included in the `delegate` maybe? What do you think?

julienbodet replied:
Are you talking about `ContextMenuDelegate`? Maybe we can modify its functions signatures to add this flag:

```
public protocol ContextMenuDelegate: class {
    func contextMenuWillDismiss(viewController: UIViewController, animated: Bool, doneTapped: Bool)
    func contextMenuDidDismiss(viewController: UIViewController, animated: Bool, doneTapped: Bool)
}
```

BasThomas replied:
Yes, that sounds good!

julienbodet commented:
>@julienbodet still on this? ...

@Huddie yep still on it. I addressed most of comments but I didn’t push yet.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
